### ISSUE TITLE
testng-failed.xml includes passed tests (6.9.10)

### ISSUE THREAD

**Issue by smccarthy** [cbeust/testng#970]
6.9.10 seems to create a wrong testng-failed.xml file. 6.9.9 seems to work correctly.

Test Class:
```
import org.testng.annotations.Test;
import static org.testng.Assert.fail;

public class FailedTestSuiteTest {
	@Test
	public void testFail() throws Exception {
		fail("FAIL THIS TEST");
	}
	
	@Test
	public void testPass() throws Exception {
	}
}
```
6.9.10 Creates a testng-failed.xml file :
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite guice-stage="DEVELOPMENT" name="Failed suite [Default suite]">
  <test verbose="2" name="Default test">
    <classes>
      <class name="com.mindflash.unused.FailedTestSuiteTest"/>
    </classes>
  </test> <!-- Default test -->
  <test name="Default test(failed)">
    <classes>
      <class name="com.mindflash.unused.FailedTestSuiteTest">
        <methods>
          <include name="testFail"/>
        </methods>
      </class> <!-- com.mindflash.unused.FailedTestSuiteTest -->
    </classes>
  </test> <!-- Default test(failed) -->
</suite> <!-- Failed suite [Default suite] -->
```

6.9.9 Creates a testng-failed.xml file :
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite guice-stage="DEVELOPMENT" name="Failed suite [Default suite]">
  <test name="Default test(failed)">
    <classes>
      <class name="com.mindflash.unused.FailedTestSuiteTest">
        <methods>
          <include name="testFail"/>
        </methods>
      </class> <!-- com.mindflash.unused.FailedTestSuiteTest -->
    </classes>
  </test> <!-- Default test(failed) -->
</suite> <!-- Failed suite [Default suite] -->
```

When I run the 6.9.10 testng-failed.xml file, I see 2 Failed and 1 Passed . I was expecting 1 failed and 0 passed. When I run the 6.9.9 testng-failed.xml file, I see 1 failed.

**Comment by testphreak** [comment]
Issue was fixed by @krmahadevan and should be in 6.9.11. https://github.com/cbeust/testng/issues/960

**Comment by juherr** [comment]
Duplicate of #960
